azure-cognitiveservices-vision-computervision/README.md

```diff
@@ -0,0 +1,287 @@
+max_descriptions=3
+
+analysis = client.describe_image(url, max_descriptions, language)
+
+for caption in analysis.captions:
+    print(caption.text)
+    print(caption.confidence)
+```
+
+### Generate thumbnail
+
+You can generate a thumbnail (JPG) of an image. The thumbnail does not need to be in the same proportions as the original image. 
+
+```Python
+from PIL import Image
+import io
+
+width=50
+height=50
+url = "http://www.public-domain-photos.com/free-stock-photos-4/travel/san-francisco/golden-gate-bridge-in-san-francisco.jpg"
+
+thumbnail = client.generate_thumbnail(width, height, url)
+
+for x in thumbnail:
+    image = Image.open(io.BytesIO(x))
+image.save('thumbnail.jpg')
+```
+
+### Get subject domain list
+
+Review the subject domains used to analyze your image. These domain names are used when [analyzing an image by domain](#analyze-an-image-by-domain). An example of a domain is `landmarks`.
+
+```Python
+models = client.list_models()
+
+for x in models.models_property:
+    print(x)
+```
+
+
+## Troubleshooting
+
+### General
+
+When you interact with the [ComputerVisionAPI][ref_computervisionclient] client using the Python SDK, errors returned by the service correspond to the same HTTP status codes returned for REST API requests:
+
+[HTTP Status Codes for Azure ComputerVisionAPI][computervision_http_status_codes]
+
+For example, if you try to analyze an image with an invalid key, a `401` error is returned. In the following snippet, the error is handled gracefully by catching the exception and displaying additional information about the error.
+
+```Python
+
+domain = "landmarks"
+url = "http://www.public-domain-photos.com/free-stock-photos-4/travel/san-francisco/golden-gate-bridge-in-san-francisco.jpg"
+language = "en"
+max_descriptions=3
+
+try:
+    analysis = client.describe_image(url, max_descriptions, language)
+
+    for caption in analysis.captions:
+        print(caption.text)
+        print(caption.confidence)
+except HTTPFailure as e:
+    if e.status_code == 401:
+        print("Error unauthorized. Make sure your key and region are correct.")
+    else:
+        raise
+```
+
+### Handle transient errors with retries
+
+While working with the [ComputerVisionAPI][ref_computervisionclient] client , you might encounter transient failures caused by [rate limits][computervision_request_units] enforced by the service, or other transient problems like network outages. For information about handling these types of failures, see [Retry pattern][azure_pattern_retry] in the Cloud Design Patterns guide, and the related [Circuit Breaker pattern][azure_pattern_circuit_breaker].
+
+## Next steps
+
+### More sample code
+
+Several Computer Vision Python SDK samples are available to you in the SDK's GitHub repository. These samples provide example code for additional scenarios commonly encountered while working with Computer Vision:
+
+* [`examples.py`][sample_examples_misc] - Contains the code snippets found in this article.
```

Comment:
Reminder that this section will need an update to point to any samples you'd like to highlight.

Reply:
That is, I suppose, unless *everyone in the Python world* knows what PIL is and that you need Pillow. I hadn't used it before, and because the [PIL package](https://pypi.org/project/PIL/) itself seems to not have a straight `pip install` story, I had to poke around a bit to find out what/how to install to get the PIL import working.

Reply:
@annatisch Is Pillow the correct/best package for this?